### ISSUE TITLE
1.1.0-pre4 compile error on FreeBSD 10.3

### ISSUE THREAD

**Issue by Sp1l** [openssl/openssl#888]
Building the beta fails on FreeBSD 10.3-RC1 / clang 3.4.

Build error for the FreeBSD port (works fine with PRE3)
```
CC="cc" /usr/local/bin/perl5 crypto/sha/asm/sha512-x86_64.pl elf crypto/sha/sha512-x86_64.s
cc -DDSO_DLFCN -DHAVE_DLFCN_H -DOPENSSL_THREADS -DOPENSSL_NO_STATIC_ENGINE -DOPENSSL_PIC -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DMD5_ASM -DAES_ASM -DVPAES_ASM -DBSAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DPOLY1305_ASM -DOPENSSLDIR="\"/usr/local/openssl\"" -DENGINESDIR="\"/usr/local/lib/engines\"" -DL_ENDIAN -Wall -O3 -pthread -D_THREAD_SAFE -D_REENTRANT -fPIC -Iinclude -I. -Icrypto/include -MMD -MF crypto/engine/eng_cryptodev.d.tmp -MT crypto/engine/eng_cryptodev.o -c -o crypto/engine/eng_cryptodev.o crypto/engine/eng_cryptodev.c
crypto/engine/eng_cryptodev.c:1465:15: error: incomplete definition of type 'struct DSA_SIG_st'
        dsaret->r = r;
        ~~~~~~^
include/openssl/dsa.h:124:16: note: forward declaration of 'struct DSA_SIG_st'
typedef struct DSA_SIG_st DSA_SIG;
               ^
crypto/engine/eng_cryptodev.c:1466:15: error: incomplete definition of type 'struct DSA_SIG_st'
        dsaret->s = s;
        ~~~~~~^
include/openssl/dsa.h:124:16: note: forward declaration of 'struct DSA_SIG_st'
typedef struct DSA_SIG_st DSA_SIG;
               ^
crypto/engine/eng_cryptodev.c:1500:23: error: incomplete definition of type 'struct DSA_SIG_st'
    if (bn2crparam(sig->r, &kop.crk_param[5]))
                   ~~~^
include/openssl/dsa.h:124:16: note: forward declaration of 'struct DSA_SIG_st'
typedef struct DSA_SIG_st DSA_SIG;
               ^
crypto/engine/eng_cryptodev.c:1502:23: error: incomplete definition of type 'struct DSA_SIG_st'
    if (bn2crparam(sig->s, &kop.crk_param[6]))
                   ~~~^
include/openssl/dsa.h:124:16: note: forward declaration of 'struct DSA_SIG_st'
typedef struct DSA_SIG_st DSA_SIG;
               ^
4 errors generated.
*** [crypto/engine/eng_cryptodev.o] Error code 1

make[2]: stopped in /usr/ports/security/openssl-devel/work/openssl-1.1.0-pre4
1 error
```
$ clang -v
FreeBSD clang version 3.4.1 (tags/RELEASE_34/dot1-final 208032) 20140512
Target: x86_64-unknown-freebsd10.3
Thread model: posix
Selected GCC installation:

**Comment by mattcaswell** [comment]
I believe the fix for this is already in git (43c1fd6b648), but unfortunately just missed getting into the beta.

**Comment by Sp1l** [comment]
Correct, that fixes the issue. Should've looked first...
